test/controllers/activestorage/blobs/proxy_controller_test.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'test_case'
+
+class ActiveStorage::Blobs::ProxyControllerTest < ActionDispatch::IntegrationTest
+  include Rails.application.routes.url_helpers
+
+  def setup
+    @user = create(:public_health_user)
+    @download = create(:download, user_id: @user.id)
+    @download.exports.attach(io: StringIO.new('text'), filename: 'text.txt', content_type: 'application/text')
+    @download_url = rails_storage_proxy_path(@download.exports.first, only_path: true)
+  end
+
+  test 'user that requested export can download' do
+    sign_in @user
+    get @download_url
+    assert_response :success
+  end
+
+  test 'user that did not request export cannot download' do
+    other_user = create(:public_health_user)
+    sign_in other_user
+    get @download_url
+    assert_redirected_to '/'
+  end
+
+  test 'non-logged in user cannot download export' do
+    get @download_url
+    assert_redirected_to '/'
```

Comment:
would non-logged in user still be rejected if somehow `@download = create(:download, user_id: nil)`?